cmd/prometheus/main.go

```diff
@@ -154,6 +156,11 @@ func Main() int {
 	webHandler := web.New(&cfg.web)
 	go webHandler.Run()
 
+	// Monitor outgoing connections with conntrack
+	http.DefaultTransport.(*http.Transport).DialContext = conntrack.NewDialContextFunc(
```

Comment:
Most connections of interest aren't made with the default transport, look at NewClientFromConfig in util/httputil and at the discovery directory.